packages/playwright-core/src/cli/program.ts

```diff
@@ -70,6 +70,7 @@ commandWithOpenOptions('codegen [url]', 'open page and generate code for user ac
       ['-o, --output <file name>', 'saves the generated script to a file'],
       ['--target <language>', `language to generate, one of javascript, playwright-test, python, python-async, python-pytest, csharp, csharp-mstest, csharp-nunit, java, java-junit`, codegenId()],
       ['--test-id-attribute <attributeName>', 'use the specified attribute to generate data test ID selectors'],
+      ['--user-data-dir <directory>', 'use the specified user data directory instead of a new context'],
```

Comment:
I think this should go to `commandWithOpenOptions` instead, so we can supply user data dir to `open`, `codegen` and other commands.